### PR TITLE
Incorporate BLEU Score, fixes to evaluator and data fetcher

### PR DESCRIPTION
**Pull request by achyudhk**
### ## Description
* Fix data fetcher to work with test splits
* Add util functions to calculate BLEU score
* Add evaluator that computes BLEU on the test split

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

**Comment by achyudhk** [comment]
Work is still on-going on replacing one-got encoding with continuous vector representations due to the following unresolved tasks on my side:

### To-do:
- [ ] Declare OOV tokens based on the word2vec vocab rather than the training set's word index
- [ ] Tokenize using NLTK rather than split()
- [ ] Find a workaround for Keras metric API limitation for calculating BLEU score while training
